sandy/gls.py

```diff
@@ -51,13 +51,39 @@ def gls_update(y, S, Vx_calc, Vy_extra, y_calc, y_extra, threshold=None):
     1   1.25714e+00
     dtype: float64
     """
-    y_ = pd.Series(y).values
-    index = pd.Series(y).index
-    y_calc_ = pd.Series(y_calc).values
-    y_extra_ = pd.Series(y_extra).values
+    # Model calculus:
+    y_calc_ = _y_calc(x_prior, S)
+    # Data in a appropiate format
+    x_prior_ = pd.Series(x_prior)
+    y_extra_ = pd.Series(y_extra)
+    y_calc_ = y_calc_.reindex(y_extra_.index)
```

Comment:
I am not sue about the reindex...the user should provide correct inidces

Reply:
The user should provide correct index. If dont pass correct index, the program will broke, avoiding the user to obtain results without sense in the best of the cases